labelCloud/io/labels/kitti.py

```diff
@@ -20,72 +23,130 @@ def __init__(
+                    if not calib_path.is_file():
+                        logging.exception(
+                            f"There is no calibration file for point cloud {pcd_path.name}."
+                            " If you want to load labels in lidar frame without transformation"
+                            " use the label format 'kitti_untransformed'."
+                            " Skipping the loading of labels for this point cloud ..."
+                        )
+                        return []
+
+                    self._calc_transforms(calib_path)
+                    xyz1 = np.insert(np.asarray(centroid), 3, values=[1])
+                    xyz1 = self.T_c2v @ xyz1
+                    centroid = tuple([float(n) for n in xyz1[:-1]])
                     dimensions = dimensions[2], dimensions[1], dimensions[0]
+                    centroid = (
+                        centroid[0],
+                        centroid[1],
+                        centroid[2] + dimensions[2] / 2,
+                    )  # centroid in KITTI located on bottom face of bbox
                 bbox = BBox(*centroid, *dimensions)
-                if self.transformed:
-                    bbox.set_rotations(
-                        0, 0, rel2abs_rotation(-float(line_elements[14]) + math.pi / 2)
-                    )
-                else:
-                    bbox.set_rotations(0, 0, rel2abs_rotation(float(line_elements[14])))
-                bbox.set_classname(line_elements[0])
-                labels.append(bbox)
+                rotation = -float(meta["rotation_y"]) + math.pi / 2 if self.transformed else float(meta["rotation_y"])
+                bbox.set_rotations(0, 0, rel2abs_rotation(rotation))
+                bbox.set_classname(meta["type"])
+                bboxes.append(bbox)
             logging.info("Imported %s labels from %s." % (len(label_lines), label_path))
-        return labels
+        return bboxes
 
     def export_labels(self, bboxes: List[BBox], pcd_path: Path) -> None:
         data = str()
 
         # Labels
-        for bbox in bboxes:
+        for i, bbox in enumerate(bboxes):
             obj_type = bbox.get_classname()
             centroid = bbox.get_center()
-            if self.transformed:
-                centroid = (-centroid[1], centroid[2] + 2.3, centroid[0])
-            location = " ".join([str(self.round_dec(v)) for v in centroid])
             dimensions = bbox.get_dimensions()
             if self.transformed:
-                dimensions = (dimensions[2], dimensions[1], dimensions[0])
+                centroid = (
+                    centroid[0],
+                    centroid[1],
+                    centroid[2] - dimensions[2] / 2,
+                )  # centroid in KITTI located on bottom face of bbox
+                dimensions = dimensions[2], dimensions[1], dimensions[0]
+                xyz1 = np.insert(np.asarray(centroid), 3, values=[1])
+                xyz1 = self.T_v2c @ xyz1
```

Comment:
Here it could theoretically happen that `self.T_v2c` was not initialized (e.g. when there was no label in the first place but now we are writing new labels).

I guess we can just use a dynamic getter instead:
- [ ] use getter function for transform